### ISSUE TITLE
HF0.2, cli_wallet create_account_delegated does not work

### ISSUE THREAD

**Issue by gropox** [GolosChain/golos#279]
*create_account_delegated* does not work

```
testnet@v22017084506852290:~$ sh wallet -s wss://ws.testnet.golos.io
locked >>> unlock 123
unlocked >>> create_account_delegated cyberfounder "1.000 GOLOS" "20000.000000 GESTS" foobar100 "{}" true
create_account_delegated cyberfounder "1.000 GOLOS" "20000.000000 GESTS" foobar100 "{}" true
....
0 exception: unspecified
10 basic_exception: tion
a.get_decimals() == b.get_decimals(): 
    {}
    th_a  asset.hpp:261 operator+

    {"op":["account_create_with_delegation",{"fee":"1.000 GOLOS","delegation":"20000.000000 GESTS","creator":"cyberfounder","new_account_name":"foobar100","owner":{"weight_threshold":1,"account_auths":[],"key_auths":[["GLS5Kx3kmbvtoUZmeFeahuPeTzFXpWQTXoiNmKmz2z9R9EZVx8J4d",1]]},"active":{"weight_threshold":1,"account_auths":[],"key_auths":[["GLS72Ad3R2SBN3iY9JB7tZvzJW7vrnGsmpxjKkH7CV51Uk4dxPDXd",1]]},"posting":{"weight_threshold":1,"account_auths":[],"key_auths":[["GLS6wjr7ric7EzjXCCEy8o98XojHrzzcCYmZJkk5HcVkaQWF7Ue98",1]]},"memo_key":"GLS7LMzZ5FhEtUuXw6gng6bNzKXm8BQAyW9DvDuf2mWxTZ6kPZPd7","json_metadata":"{}","extensions":[]}]}
    th_a  database.cpp:3089 _apply_transaction

unlocked >>> list_my_accounts
cyberfounder       43305758.000 GOLOS 2080395.518546 GESTS            0.000 GBG
-------------------------------------------------------------------------
TOTAL              43305758.000 GOLOS 2080395.518546 GESTS            0.000 GBG

```

**Comment by gropox** [comment]
It seems to be failing at this line 

https://github.com/GolosChain/golos/blob/6b2151f00c209ac14225810b94324d760e1649e8/libraries/protocol/include/golos/protocol/asset.hpp#L261

The commit above didn't fixed the problem

**Comment by Nemo1369** [comment]
@gropox Thanks for the issue. Decimals comparison is a crucial thing. It was not required when there were only two assets with the same precision. But not it is important. That is what I'm going to fix.

**Comment by gropox** [comment]
Do you mean there are cases where same assets exist with different precision? How could it be? 

Even if so, you cold takeover precision of asset with greather precision after addition, substraction. But really, I cannot image where such case is possible.

**Comment by Nemo1369** [comment]
@gropox Asset precision may vary from 1 to 12 decimals. It depends on what params were put into ```asset_create_operation```. For example, what exactly you are trying to do is to compare GESTS with 6 decimals precision and GOLOS with 3 decimals precision. And that needs to be handled.

**Comment by gropox** [comment]
But why should one compare different assets, it does not make sence. 

You wrote "But now it is important." Therefore I tought there are cases where e.g.  GOLOS compared against GBG, which useless is.

**Comment by Nemo1369** [comment]
@gropox It should not. That is why the solution is to fix business-logic, what exactly I'm going to do after #278.

**Comment by Nemo1369** [comment]
Fixed with 16f2c91e3a
